components/brave_new_tab_ui/components/default/rewards/index.tsx

```diff
@@ -128,6 +130,14 @@ export const RewardsWidget = createWidget((props: RewardsProps) => {
     chrome.braveRewards.enableAds()
   }
 
+  const walletKeys = Object.keys(props.balance.wallets)
```

Comment:
Instead of inferring the "loading" status on the front-end, can we get it directly in C++?

What if we updated the `chrome.braveRewards.fetchBalance` function definition (in `brave_rewards.json`) such that the "balance" return object is optional?

In `BraveRewardsFetchBalanceFunction`, if `result` is not `LEDGER_OK`, then we'd respond with no arguments. Then in here, we'd just have something like:

```
rewardsBalance={optional(props.balance ? props.balance.total : undefined)}
```

Reply:
The return value of the fetch balance flow is unusable, unfortunately, as the results of the intermediate fetch balance requests in the chain are discarded. For instance, if the user has a verified Uphold wallet, and the Uphold balance request fails, the overall result will still be `LEDGER_OK`, as the Gemini (which is the last) part of the chain is successful (as there's no Gemini wallet).

We need to know the type of the (connected) external wallet (if there is one) in `BraveRewardsFetchBalanceFunction` in order to be able to determine the outcome of the flow.